zcash_primitives/src/sapling/keys.rs

```diff
@@ -210,4 +359,24 @@ mod tests {
         // nk is allowed to be the identity.
         assert!(FullViewingKey::read(&buf[..]).is_ok());
     }
+
+    #[test]
+    fn dfvk_round_trip() {
+        let dfvk = {
+            let extsk = zip32::ExtendedSpendingKey::master(&[]);
+            let extfvk = zip32::ExtendedFullViewingKey::from(&extsk);
+            DiversifiableFullViewingKey::from(extfvk)
+        };
+
+        // Check value -> bytes -> parsed round trip.
+        let dfvk_bytes = dfvk.to_bytes();
+        let dfvk_parsed = DiversifiableFullViewingKey::from_bytes(&dfvk_bytes).unwrap();
+        assert_eq!(dfvk_parsed.fvk.vk.ak, dfvk.fvk.vk.ak);
+        assert_eq!(dfvk_parsed.fvk.vk.nk, dfvk.fvk.vk.nk);
+        assert_eq!(dfvk_parsed.fvk.ovk, dfvk.fvk.ovk);
+        assert_eq!(dfvk_parsed.dk, dfvk.dk);
```

Comment:
Why not derive `Eq`?

Reply:
I tend to avoid deriving `PartialEq, Eq` on key types that could be considered secret, or for which a comparison doesn't really make sense in the public API. In those situations, reaching into the internals is fine. But if it is useful in the public API, someone could add a `PartialEq` implementation backed by `ConstantTimeEq`.